src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1927,16 +2016,32 @@ impl Cage {
         0 //_close_helper has succeeded!
     }
 
-    //------------------------------------FCNTL SYSCALL------------------------------------
     
-    //fcntl performs operations, like returning or setting file status flags,
-    //duplicating a file descriptor, etc., on an open file descriptor 
-    //it accepts three parameters: fd - an open file descriptor, cmd - an operation to be performed on fd,
-    //and arg - an optional argument (whether or not arg is required is determined by cmd)
-    //for a successful call, the return value depends on the operation and can be one of: zero, the new file descriptor, 
-    //value of file descriptor flags, value of status flags, etc.
-    //for more detailed description of all the commands and return values, see 
-    //https://linux.die.net/man/2/fcntl
+    /// ### Description
+    /// 
+    /// `fcntl_syscall` performs operations, like returning or setting file status flags,
+    /// duplicating a file descriptor, etc., on an open file descriptor 
+    /// 
+    /// ### Arguments
+    /// 
+    /// it accepts three parameters: 
+    /// * `fd` - an open file descriptor
+    /// * `cmd` - an operation to be performed on fd
+    /// * `arg` - an optional argument (whether or not arg is required is determined by cmd)
+    /// 
+    /// ### Returns
+    /// 
+    /// for a successful call, the return value depends on the operation and can be one of: zero, the new file descriptor, 
+    /// value of file descriptor flags, value of status flags, etc.
+    /// 
+    /// ### Errors
+    /// 
+    /// * EBADF - fd is not a valid file descriptor
+    /// * EINVAL - doesnt match implementation parameters
+    /// 
```

Comment:
I think it's typical to have a 'panic' block as well, which lists all the instances in which the code panics.

Reply:
will update it.